car.js

```diff
@@ -3,26 +3,37 @@
 class Car {
 
   start_car(options) {
-    if (options && options['headlights'])
+    console.log(options);
+    console.log(this);
+    if (this.hasOwnProperty('headlights')){
       this.headlights = options['headlights']
-    else
+    } else {
       this.headlights = true
+    }
 
-    if (options && options['gear'])
+    if (this.hasOwnProperty('gear')){
       this.gear = options['gear']
-    else
+    } else {
       this.gear = 1
+    }
   }
 }
 
-var herbie = new Car
+var herbie = new Car()
 
 // When we start the car, the headlights should be on by default
 herbie.start_car()
 console.log(herbie.headlights) // true
+console.log(herbie.gear);
 
 // But what happens when we want to start without headlights?
+
+herbie.start_car({headlights : false, gear: 2})
+console.log(herbie.headlights) // true ...KOK BISA?!
+console.log(herbie.gear);
+
 herbie.start_car({
-  headlights: false
+  gear: 3
 })
-console.log(herbie.headlights) // true ...KOK BISA?!
+console.log(herbie.headlights)
```

Comment:
itu buat memperlihatkan kalau this dijavascript masih nyangkut, ada proses memory yang tersisa saat tapi saat dipanggil ngga ada